### PR TITLE
Defunt child proc

### LINKED ISSUE #745: sys.exec disown defunc prcoess
**Describe the bug**
When `sys.exec` uses the disown flag on Linux a defunc child process is left hanging indefinitely.

image

**To Reproduce**
```python3
def main():
    sys.exec("/bin/sleep", ["10"], True)
    print(sys.shell("ps -faux")['stdout'])
    time.sleep(30)

main()
```

```bash
root ➜ /workspaces/realm/implants/golem (defunt-child-proc) $ cargo run -- /tmp/test.eldritch 
```

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix defunct child processes

#### Which issue(s) this PR fixes:
Fixes #745

### CODE REVIEW

File: implants/lib/eldritch/src/sys/exec_impl.rs [hunk 1]
```diff
@@ -66,7 +68,7 @@ fn handle_exec(path: String, args: Vec<String>, disown: Option<bool>) -> Result<
             }
             ForkResult::Child => {
                 setsid()?;
-                match unsafe { fork()? } {
+                match unsafe { nix::unistd::fork()? } {
```
Comment (Cictrone) [inline implants/lib/eldritch/src/sys/exec_impl.rs:71]: i think this is the same as just `fork()?` right?

File: implants/lib/eldritch/src/sys/exec_impl.rs [hunk 2]
```diff
@@ -183,6 +227,30 @@ mod tests {
...
+            || cfg!(target_os = "freebsd")
+            || cfg!(target_os = "openbsd")
+            || cfg!(target_os = "netbsd")
+        {
+            let _res = handle_exec(
+                String::from("/bin/sleep"),
+                vec!["3".to_string()],
+                Some(true),
+            )?;
+            let res = get_zombie_child_processes(process::id())?;
```
Comment (Cictrone) [inline implants/lib/eldritch/src/sys/exec_impl.rs:248]: should you wait 3 seconds here?

**Review by Cictrone — DISMISSED** [review]
two minor nits, also please make sure no lint issues 😎

File: implants/lib/eldritch/src/sys/exec_impl.rs (hunk 2 above)
Comment (hulto) [reply implants/lib/eldritch/src/sys/exec_impl.rs:250]: The issue we were seeing was that the intermediate process was hanging in a defunct state not the grandchild.

**Review by Cictrone — APPROVED** [review]

**Pull request merged**